spec/integration/aws_internet_gateway_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe Chef::Resource::AwsInternetGateway do
+  extend AWSSupport
+
+  when_the_chef_12_server "exists", organization: 'foo', server_scope: :context do
+    with_aws "with a VPC and an internet gateway" do
+      vpc = nil
+      internet_gateway = nil
+
+      before {
```

Comment:
Instead of this before block you should use

```ruby
aws_vpc "test_internet_gateway" do
  cidr_block "10.0.0.0/24"
end

# Later when you need to reference it...
test_internet_gateway.aws_object
```

Then VPC will get automatically torn down at the end then